C++/test/algorithm/backtracking/n_queens.cpp

```diff
@@ -0,0 +1,49 @@
+#include "third_party/catch.hpp"
+#include "algorithm/backtracking/n_queens.hpp"
+
+TEST_CASE("Unplaceable cases", "[backtracking][n_queens]") {
+    NQueensSolver b0(0);
+    REQUIRE_FALSE(b0.can_place_queens());
+    NQueensSolver b2(2);
+    REQUIRE_FALSE(b2.can_place_queens());
+    NQueensSolver b3(3);
+    REQUIRE_FALSE(b3.can_place_queens());
+}
+
+TEST_CASE("Placeable cases", "[backtracking][n_queens]") {
+    NQueensSolver b1(1);
+    REQUIRE(b1.can_place_queens());
```

Comment:
I know it's redundant and less concise, but since this repo is generally oriented toward less experienced programmers, I like `REQUIRE(b1.can_place_queens() == true);`. If you're unfamiliar with Catch I think this will be easier to understand.